internal/compiler/parser.rs

```diff
@@ -762,6 +762,9 @@ impl SyntaxToken {
             })?;
         Some(SyntaxToken { token, source_file: self.source_file.clone() })
     }
+    pub fn text(&self) -> &str {
```

Comment:
I hope it's ok to expose SyntaxToken's text to be able to retain whitespace between sub-elements.